tests/refs/reflog/reflog.c

```diff
@@ -154,6 +154,18 @@ void test_refs_reflog_reflog__reading_the_reflog_from_a_reference_with_no_log_re
 	git_buf_free(&subtrees_log_path);
 }
 
+void test_refs_reflog_reflog__reading_a_reflog_with_invalid_format_returns_error(void)
+{
+	git_reflog *reflog;
+	git_error *error;
+
+	cl_git_fail(git_reflog_read(&reflog, g_repo, "refs/heads/extra-newline"));
+
+	error = giterr_last();
+	cl_assert(error);
```

Comment:
If you don't clear the error state before calling `git_reflog_read()`, you can't be sure you're looking at what `git_reflog_read()` gave you.